src/main/java/com/google/api/codegen/transformer/ApiMethodTransformer.java

```diff
@@ -0,0 +1,276 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer;
+
+import com.google.api.codegen.MethodConfig;
+import com.google.api.codegen.ServiceMessages;
+import com.google.api.codegen.util.Name;
+import com.google.api.codegen.viewmodel.ApiMethodDocView;
+import com.google.api.codegen.viewmodel.ApiMethodType;
+import com.google.api.codegen.viewmodel.DynamicDefaultableParamView;
+import com.google.api.codegen.viewmodel.MapParamDocView;
+import com.google.api.codegen.viewmodel.OptionalArrayMethodView;
+import com.google.api.codegen.viewmodel.ParamDocView;
+import com.google.api.codegen.viewmodel.RequestObjectParamView;
+import com.google.api.codegen.viewmodel.SimpleParamDocView;
+import com.google.api.tools.framework.model.Field;
+import com.google.api.tools.framework.model.TypeRef;
+import com.google.protobuf.DescriptorProtos.FieldDescriptorProto.Type;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * ApiMethodTransformer generates view objects from method definitions.
+ */
+public class ApiMethodTransformer {
+  private InitCodeTransformer initCodeTransformer;
+
+  public ApiMethodTransformer() {
+    this.initCodeTransformer = new InitCodeTransformer();
```

Comment:
Is there a deliberate reason to initialize here rather than at the point of field declaration? I generally prefer the latter unless there's complex logic or it depends on a constructor parameter, but it's a weak preference.